### PR TITLE
Luacheck things

### PR DESCRIPTION
**Pull request by wsor4035**
.luacheckrc file improvements, adds github workflow, tosses ancient settings no one uses.

### CODE REVIEW

File: .github/workflows/luacheck.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+name: luacheck
+on: [push, pull_request]
+jobs:
+  build:
```
Comment (S-S-X) [inline .github/workflows/luacheck.yml:4]: Suggestion to use better name for luacheck job, maybe luacheck or lint. This is different from "name: luacheck" and is used to distinguish different workflow jobs for GitHub task selection in checks and such.
```suggestion
  luacheck:
```

File: dryplants/juncus.lua [hunk 2]
```diff
@@ -105,8 +105,8 @@ biome_lib.register_on_generate({
 		"sumpf:peat",
 		"sumpf:sumpf"
 	},
-    max_count = JUNCUS_NEAR_WATER_PER_MAPBLOCK,
-    rarity = 101 - JUNCUS_NEAR_WATER_RARITY,
+    max_count = 70,
```
Comment (Panquesito7) [inline dryplants/juncus.lua:108]: Shouldn't we create settings for this using `settingtypes.txt`, just for compatibility?

Comment (Niklp09) [reply dryplants/juncus.lua:108]: I don't think the extra effort is justified by the number of users who use the settings

Comment (wsor4035) [reply dryplants/juncus.lua:108]: > Shouldn't we create settings for this using `settingtypes.txt`, just for compatibility?

no. see above

Comment (wsor4035) [reply dryplants/juncus.lua:108]: besides, once converted off biome lib, the settings would have been most likely removed anyways

File: trunks/nodes.lua [hunk 3]
```diff
@@ -239,98 +239,93 @@ minetest.register_node("trunks:twigs_roof_corner_2", {
...
-			then
-				minetest.swap_node(pos, {name=corner, param2=0})
-			end
+local roof = "trunks:twigs_roof"
+local corner = "trunks:twigs_roof_corner"
+local corner_2 = "trunks:twigs_roof_corner_2"
+
+minetest.register_abm({
+	nodenames = {roof},
+	interval = 1,
```
Comment (Niklp09) [inline trunks/nodes.lua:248]: Side note: maybe we can lower the interval a bit

**Review by Niklp09 — APPROVED** [review]
seems to be ok

File: .luacheckrc [hunk 4]
```diff
@@ -1,14 +1,21 @@
 unused_args = false
 allow_defined_top = true
+max_line_length = 185
+
+exclude_files = {".luacheckrc"}
```
Comment (OgelGames) [inline .luacheckrc:5]: This isn't needed, `.luacheckrc` is not a `.lua` file, so it's already ignored :P

**Pull request merged**